WordPress/openverse
# `mature` -> `sensitive` frontend copy and code

## Fixes



Fixes #2623 by @AetherUnbound 
Fixes #2624 by @AetherUnbound 

Also related to the overall #377 project as the new feature uses "sensitive" across the board and the content report modal is really the last place that was still using the legacy term "mature".

## Description


After doing the copy changes first, the code changes were so minimal (I think because a lot of things have been changed during the sensitive content feature implementation since the implementation plan for these two issues was written) that it did not make sense to separate them. If they were separated, there would be weird decisions like leaving the `mature` key in `en.json5` even though the copy itself was already updated. If we really want to separate them, I can open two separate PRs, but I don't see the value in it anymore since the scope of these issues has so significantly reduced since the implementation plan was written.

The only "dangling" bit of code changes is caused by the Openverse API only accepting "mature" for the report reason.

To handle that, I've created one commit that maps the reason on the frontend to `mature`. That is a standalone commit. However, because it seemed just as straightforward to update the API to accept "sensitive" as the reason, and map it API side, I've added a subsequent commit that removes the mapping on the frontend, and updates the API to map "sensitive" report reason to "mature" (which the database expects). If that's too much to include in this PR and it would be preferable to just map it on the frontend, reviewers, please let me know, and we can revert the commit with API changes.

## Testing Instructions

Checkout the branch and run `just p frontend dev`. Visit the frontend and confirm the content report modal looks good and the options work. That's the only real copy change here. The rest are code changes. Unit tests and e2e tests should all pass. Double check the updated visual regression snapshots to confirm everything there makes sense as well.

## Checklist


- [N/A] I added or updated documentation (if applicable).
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
